yarn-project/foundation/src/abi/abi.ts

```diff
@@ -404,18 +409,32 @@ export function getFunctionDebugMetadata(
   contractArtifact: ContractArtifact,
   functionArtifact: FunctionArtifact,
 ): FunctionDebugMetadata | undefined {
-  if (functionArtifact.debugSymbols && contractArtifact.fileMap) {
-    const programDebugSymbols = JSON.parse(
-      inflate(Buffer.from(functionArtifact.debugSymbols, 'base64'), { to: 'string', raw: true }),
-    );
-    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/5813)
-    // We only support handling debug info for the contract function entry point.
-    // So for now we simply index into the first debug info.
-    return {
-      debugSymbols: programDebugSymbols.debug_infos[0],
-      files: contractArtifact.fileMap,
-    };
+  try {
+    if (functionArtifact.debugSymbols && contractArtifact.fileMap) {
+      // TODO(https://github.com/AztecProtocol/aztec-packages/issues/10546) investigate why debugMetadata is so big for some tests.
+      const programDebugSymbols = JSON.parse(
+        inflate(Buffer.from(functionArtifact.debugSymbols, 'base64'), { to: 'string', raw: true }),
+      );
+      // TODO(https://github.com/AztecProtocol/aztec-packages/issues/5813)
+      // We only support handling debug info for the contract function entry point.
+      // So for now we simply index into the first debug info.
+      return {
+        debugSymbols: programDebugSymbols.debug_infos[0],
+        files: contractArtifact.fileMap,
+      };
+    }
+  } catch (err: any) {
+    if (err instanceof RangeError && err.message.includes('Invalid string length')) {
+      logger.warn(
+        `Caught RangeError: Invalid string length. This suggests the debug_symbols field of the contract ${contractArtifact.name} and function ${functionArtifact.name} is huge; too big to parse. We'll skip returning this info until this issue is resolved. Here's the error:\n${err.message}`,
+      );
+      // We'll return undefined.
+    } else {
+      // Rethrow unexpected errors
+      throw err;
+    }
```

Comment:
For some reason, the debug_symbols field of two of the functions in the `StatefulTest` contract became _huge_, and when the tests tried to grab that function's artifact, it wouldn't be able to parse it (too big). See the `inflate` function being called in the function directly below this one.
See https://github.com/AztecProtocol/aztec-packages/issues/10546
Anyway, this fix only assigns debugMetadata if it can be parsed.